internal/controller/progressive/progressive.go

```diff
@@ -494,13 +492,18 @@ func startUpgradeProcess(
 	if err != nil {
 		return newUpgradingChildDef, false, err
 	}
+	// reset the Status to reflect our new Upgrading child
+	err = rolloutObject.ResetUpgradingChildStatus(newUpgradingChildDef)
+	if err != nil {
+		return newUpgradingChildDef, false, fmt.Errorf("processing upgrading child, failed to reset the upgrading child status for child %s/%s", newUpgradingChildDef.GetNamespace(), newUpgradingChildDef.GetName())
+	}
```

Comment:
We were previously calling this too late in the game (in [processUpgradingChild()](https://github.com/numaproj/numaplane/blob/main/internal/controller/progressive/progressive.go#L242-L252)) - now reset the upgrading status as soon as we have a new upgrading child. Note, I left the other call to this in there just in case it doesn't get set here.